hs/src/Reach/Eval/Core.hs

```diff
@@ -4957,14 +4963,11 @@ doToConsensus ks (ToConsensusRec {..}) = locAt slptc_at $ do
               , st_after_first = True
               }
       setSt st_recv
-      mapM_ (flip trackToken Nothing) tok_as
+      trackTokens $ map (,Nothing) toks
```

Comment:
Why did you change it to take a list? It seems way more complicated now?

Reply:
For `toConsensus` we can publish many tokens, and I thought that processing the distinct check for all the tokens at once would be more efficient. I can switch it back to handle one token at a time though

Reply:
I think it will be the same amount of work but in a different order